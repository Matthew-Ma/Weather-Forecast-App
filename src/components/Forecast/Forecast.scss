.ui.four.cards {
    min-width: 25em;
}
