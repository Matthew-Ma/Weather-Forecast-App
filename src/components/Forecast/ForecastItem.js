import React from 'react';
import { Card, Image } from 'semantic-ui-react';
import { ICON_URL } from '../../config/config';
import './ForecastItem.scss';

const ForecastItem = (props) => {
    return (
        <React.Fragment>
            <Card className="card-item">
                <Card.Content>
                    <Image
                        size="tiny"
                        src={`${ICON_URL}${props.abbreviation}.svg`}
                        centered
                    />
                    <Card.Description>{props.date}</Card.Description>
                    <Card.Meta>{props.state}</Card.Meta>
                    <Card.Meta>{Math.round(props.windSpeed)}mph</Card.Meta>
                    <Card.Meta>
                        <span>
                            {Math.round(props.tempMax)}
                            <sup>&deg;</sup>
                            <small>/</small>
                            {Math.round(props.tempMin)}
                            <sup>&deg;</sup>
                        </span>
                    </Card.Meta>
                </Card.Content>
            </Card>
        </React.Fragment>
    );
};

export default ForecastItem;
