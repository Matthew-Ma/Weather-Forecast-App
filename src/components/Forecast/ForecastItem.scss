.ui.card.card-item {
  background-color: rgba(255, 255, 255, 0.75);
  .content .description {
    color: #00b5ad;
  }
}
