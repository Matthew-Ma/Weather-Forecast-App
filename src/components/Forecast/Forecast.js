import React from 'react';
import { useSelector } from 'react-redux';
import ForecastItem from './ForecastItem';
import { Card } from 'semantic-ui-react';
import { DAYS_TO_DISPLAY } from '../../config/config';

const Forecast = () => {
    const { forecast } = useSelector((state) => ({
        forecast: state.weather,
    }));

    return (
        <Card.Group itemsPerRow={DAYS_TO_DISPLAY}>
            {forecast?.weatherData?.slice(0, DAYS_TO_DISPLAY).map((item) => {
                return (
                    <ForecastItem
                        key={item.id}
                        tempMax={item.max_temp}
                        tempMin={item.min_temp}
                        windSpeed={item.wind_speed}
                        state={item.weather_state_name}
                        abbreviation={item.weather_state_abbr}
                        date={item.applicable_date}
                    />
                );
            })}
        </Card.Group>
    );
};

export default Forecast;
