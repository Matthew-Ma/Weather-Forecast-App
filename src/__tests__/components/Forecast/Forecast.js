import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Forecast from '../../../components/Forecast/Forecast';

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

const mockData = [];
mockData.push({
    id: 1,
    max_temp: 1,
    min_temp: 1,
    wind_speed: 1,
    weather_state_name: 'word',
    weather_state_abbr: 'word',
    applicable_date: 'word',
});

describe('Forecast component:', () => {
    beforeEach(() => {
        useSelector.mockImplementation((selector) =>
            selector({
                weather: mockData,
            }),
        );
    });
    afterEach(() => {
        useSelector.mockClear();
    });
    it('renders Forecast', () => {
        const { container } = render(<Forecast />);
        expect(container).toMatchSnapshot();
    });
});
