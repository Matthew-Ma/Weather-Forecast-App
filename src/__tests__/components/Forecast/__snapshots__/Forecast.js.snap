// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast component: renders Forecast 1`] = `
<div>
  <div
    class="ui four cards"
  />
</div>
`;
